### PR TITLE
👽 Update moshi converter, retrofit, add anotation

### PR DESCRIPTION
**Pull request by RomanPolach**
Update moshi, add anotation

### CODE REVIEW

File: ass/src/main/java/cz/ackee/ass/api/AssRequest.kt [hunk 1]
```diff
@@ -1,9 +1,12 @@
 package cz.ackee.ass.api
 
+import com.squareup.moshi.JsonClass
+
 /**
  * Information that is sent for every request, mostly those that can be collected without user's
  * help. User-defined parameters are stored in [customData].
  */
+@JsonClass(generateAdapter = true)
```
Comment (davidbilik) [inline ass/src/main/java/cz/ackee/ass/api/AssRequest.kt:9]: this will still not work. You need to add kapt dependency to moshi kotlin code gen

Comment (davidbilik) [reply ass/src/main/java/cz/ackee/ass/api/AssRequest.kt:9]: and please use explicit moshi dependency

File: ass/build.gradle [hunk 2]
```diff
@@ -23,8 +23,9 @@ dependencies {
     implementation "com.android.support:design:27.1.1"
     implementation "com.android.support.constraint:constraint-layout:1.1.3"
 
-    implementation "com.squareup.retrofit2:retrofit:2.5.0"
-    implementation "com.squareup.retrofit2:converter-moshi:2.5.0"
+    implementation "com.squareup.retrofit2:retrofit:2.7.0"
+    implementation "com.squareup.retrofit2:converter-moshi:2.7.0"
+    kapt "com.squareup.moshi:moshi-kotlin-codegen:2.7.0"
```
Comment (davidbilik) [inline ass/build.gradle]: please, try to check the code that you push. or at least try to build the project. Version 2.7.0 of moshi does not exist. And add also regular moshi dependency, not just a codegen

Comment (RomanPolach) [reply ass/build.gradle]: Interesting it did not fail during sync

Comment (davidbilik) [reply ass/build.gradle]: I think for kapt you need to run the project and it will fetch the dependencies when it wants to use them

Comment (RomanPolach) [reply ass/build.gradle]: fixed

File: ass/src/main/java/cz/ackee/ass/api/AssRequest.kt (hunk 1 above)
Comment (RomanPolach) [reply ass/src/main/java/cz/ackee/ass/api/AssRequest.kt:9]: What do you mean by explicit? There is this moshi dependency

Comment (davidbilik) [reply ass/src/main/java/cz/ackee/ass/api/AssRequest.kt:9]: there is not moshi dependency, there is depenency on codegen and for retrofit converter, not for moshi itself.

**Pull request merged**